### ISSUE TITLE
1d Sample code produces scale_x_continuous error

### ISSUE THREAD

**Issue by rwbaer** [RGLab/ggcyto#8]
https://github.com/RGLab/ggcyto/blob/trunk/vignettes/advanced/ggplot.flowSet.1d.md

The following 1d sample code seems to produce an error:

...
```
> # logicle scale with customized breaks
> myBreaks <- c(0, 10, 1e2, 1e3)
> p  + scale_x_logicle(t = 1e4, w = 0, 
+                      breaks = myBreaks)
`stat_bin()` using `bins = 30`. Pick better value with
`binwidth`.
Error in if (x[i] == 0) quote(0) else if (drop.1 && mT[i] == 1) substitute(10^E,  : 
  missing value where TRUE/FALSE needed
> # remove marginal events
> p  + scale_x_logicle(t = 1e4, w = 0, 
+                      breaks = myBreaks, limits = c(2, 1e4))
`stat_bin()` using `bins = 30`. Pick better value with
`binwidth`.
Error in if (x[i] == 0) quote(0) else if (drop.1 && mT[i] == 1) substitute(10^E,  : 
  missing value where TRUE/FALSE needed
In addition: Warning message:
Removed 1136 rows containing non-finite values (stat_bin). 
```
...

**Comment by mikejiang** [comment]
It was a little tricky to troubleshoot since error occurs within the private function defined in `ggproto` object. Anyway, the root cause was the `scales::censor` function invoked by `ggplot2::ScaleContinuous$get_breaks`, which tries to flag and replace any `break points` (in this case `0`) that are outside of the data limits with `NA`.Then `NA` fails the label format function `pretty10exp`.  
The easy workaround will be switching to default `format_format()` function whenever the custom `breaks` are provided.

**Comment by mikejiang** [comment]
Actually, `ggplot` has the mechanism to hide these `out-of-bound` break point that has been replaced with `NA` by `censor` function. All I need to do is to preserve these `NA` values in our custom format function (i.e. `pretty10exp`) and let `ggplot` do its job.

**Comment by mikejiang** [comment]
@rwbaer , the fix is in `flowWorkspace 3.18.9`.
